xfer/publisher.go

```diff
@@ -52,16 +53,24 @@ func NewHTTPPublisher(target, token, id string) (*HTTPPublisher, error) {
 
 // Publish publishes the report to the URL.
 func (p HTTPPublisher) Publish(rpt report.Report) error {
-	var buf bytes.Buffer
-	if err := gob.NewEncoder(&buf).Encode(rpt); err != nil {
+	gzbuf := new(bytes.Buffer)
+	gzwriter := gzip.NewWriter(gzbuf)
```

Comment:
Also, I think we could use a io.Pipe in a way such that we don't buffer the whole encoded report in memory.  Give one end to http.NewRequest, and the other to the gob.NewEncoder.  Not 100% sure, but can you give it a try please?